azkaban-common/src/main/java/azkaban/scheduler/TriggerBasedScheduleLoader.java

```diff
@@ -83,7 +83,7 @@ private Condition createTriggerCondition(Schedule s) {
     ConditionChecker checker =
         new BasicTimeChecker("BasicTimeChecker_1", s.getFirstSchedTime(),
             s.getTimezone(), s.isRecurring(), s.skipPastOccurrences(),
-            s.getPeriod());
+            s.getPeriod(), s.getCronExpression());
```

Comment:
These two methods seem to be identical except the names. Any idea why two methods are needed?